File: game-db/src/main/java/com/jacey/game/db/entity/PlayUserEntity.java
package com.jacey.game.db.entity;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;

/**
 * @Description: 玩家信息
 * @Author: JaceyRuan
 * @Email: jacey.ruan@outlook.com
 */
@Data
@Document(collection = PlayUserEntity.COLLECTION_NAME)
public class PlayUserEntity {

    public static final String COLLECTION_NAME = "play_user";

    @Id
    private int userId;

    /** 玩家名称 */
    private String username;

    /** 玩家昵称 */
    private String nickname;

    /** MD5加密密码 */
    private String passwordMD5;

    /** 注册时间戳 */
    private Date registTimestamp;

    /** 注册ip */
    private String registIp;

    /** 最后一次登录时间 */
    private Date lastLoginTimestamp;

    /** 最后一次登录ip */
    private String lastLoginIp;
}

File: game-db/src/main/kotlin/com/jacey/game/db/entity/PlayUserEntity.kt
package com.jacey.game.db.entity

import com.jacey.game.common.framework.mongo.DbDocument
import java.util.Date

/**
 * 玩家信息（Mongo 集合 play_user）
 * 字段与原 PlayUserEntity 完全一致，集合名不变。
 */
data class PlayUserEntity(
    override val _id: Int,
    /** 玩家名称 */
    var username: String? = null,
    /** 玩家昵称 */
    var nickname: String? = null,
    /** MD5加密密码 */
    var passwordMD5: String? = null,
    /** 注册时间戳 */
    var registTimestamp: Date? = null,
    /** 注册ip */
    var registIp: String? = null,
    /** 最后一次登录时间 */
    var lastLoginTimestamp: Date? = null,
    /** 最后一次登录ip */
    var lastLoginIp: String? = null,
) : DbDocument<Int> {
    val userId: Int get() = _id

    companion object {
        const val COLLECTION_NAME = "play_user"
    }
}